### PR TITLE
Fix dns discovery issue 

### PR DESCRIPTION
**Pull request by matkt**
Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

## PR description

We are creating but not starting the dns discovery daemon. After this modification the node find peers instantly.

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java
Comment (garyschulte) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java:227]: :+1:  amazing that this has been missing since it was originally implemented

Comment (garyschulte) [reply ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java:227]: we should add a test that ensures dnsDaemon is started if dnsDiscoveryURL is present

**Review by garyschulte — CHANGES_REQUESTED** [review]
great find.  I think given the changes in Diego's PR #2944, we should add a test.

**Review by garyschulte — APPROVED** [review]
I am approving to unblock the PR, but needs approval from another contributor

**Review by macfarla — APPROVED** [review]
LGTM

**Pull request merged**
